receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -65,6 +65,83 @@ func (ghs *githubScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
 	currentDate := time.Now().Day()
 	ghs.logger.Sugar().Debugf("current date: %v", currentDate)
 
-	ghs.logger.Sugar().Debug("github scraper has started")
-	return ghs.mb.Emit(), nil
+	genClient, restClient, err := ghs.createClients()
+	if err != nil {
+		ghs.logger.Sugar().Errorf("unable to create clients", zap.Error(err))
+	}
+
+	// Do some basic validation to ensure the values provided actually exist in github
+	// prior to making queries against that org or user value
+	exists, ownertype, err := ghs.checkOwnerExists(ctx, genClient, ghs.cfg.GitHubOrg)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("error checking if owner exists", zap.Error(err))
+	}
+
+	typeValid, err := checkOwnerTypeValid(ownertype)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("error checking if owner type is valid", zap.Error(err))
+	}
+
+	if !exists || !typeValid {
+		ghs.logger.Sugar().Error("error logging in and getting data from github")
+		return ghs.mb.Emit(), err
+	}
+
+	// Generate the search query based on the type, org/user name, and the search_query
+	// value if provided
+	sq := genDefaultSearchQuery(ownertype, ghs.cfg.GitHubOrg)
+
+	if ghs.cfg.SearchQuery != "" {
+		sq = ghs.cfg.SearchQuery
+		ghs.logger.Sugar().Debugf("using search query where query is: %v", ghs.cfg.SearchQuery)
+	}
+
+	// Get the repository data based on the search query retrieving a slice of branches
+	// and the recording the total count of repositories
+	repos, count, err := ghs.getRepos(ctx, genClient, sq)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("error getting repo data", zap.Error(err))
+		return ghs.mb.Emit(), err
+	}
+
+	ghs.mb.RecordGitRepositoryCountDataPoint(now, int64(count))
+
+	// Get the branch count (future branch data) for each repo and record the given metrics
+	max := 5
```

Comment:
what is max here? Is it the max number of repositories?

Reply:
max is the max number of processing in parallel for the buffered chan.

Reply:
there is some complexity here, maybe just do one worker for now and see how we can parallelize in a second iteration.

Reply:
i opted for just leveraging wait group and not doing a buffered channel.